File: 2023/Leetcode/problem1047.py
class Solution:
    def removeDuplicates(self, s: str) -> str:
        myStack = []
        for char in s:
            if len(myStack) != 0 and char == myStack[-1]:
                myStack.pop()
            else:
                myStack.append(char)
            
        return ''.join(myStack)

File: 2024/Leetcode/problem1047.js
/**
 * @param {string} s
 * @return {string}
 */
var removeDuplicates = function(s) {
    let resultArr = [s[0]];

    for (let i = 1; i < s.length; i++){
        resultArr.push(s[i]);

        if (resultArr.length >= 2 && resultArr[resultArr.length - 1] === resultArr[resultArr.length - 2])
        {
            resultArr.pop();
            resultArr.pop();
        }
    }
    let resultString = resultArr.join("");

    return resultString;
};